### PR TITLE
refactor(api-markdown-documenter): Unify "file-safe-name" helper utilities

### PR DESCRIPTION
**Pull request by Josmithr**
Refactors a small collection of utility functions aimed at generating consistent, unique, and file-name-safe identifiers for API items (used for naming files, heading IDs, etc.). Unifies all of the exported functions aimed at providing this functionality into a single one, which also now captures package-name policy (previously duplicated in a few places).

Breaking change: publicly exposed function is now called `getFileSafeNameForApiItem`

### CODE REVIEW

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/DocumentationSuite.ts [hunk 1]
```diff
@@ -276,20 +274,15 @@ export namespace DefaultDocumentationSuiteOptions {
...
 		switch (itemKind) {
 			case ApiItemKind.Model: {
 				return "index";
 			}
-			case ApiItemKind.Package: {
-				return getSafeFilenameForName(getUnscopedPackageName(apiItem as ApiPackage));
-			}
 			default: {
-				return getQualifiedApiItemName(apiItem);
+				return getFileSafeNameForApiItem(apiItem);
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/configuration/DocumentationSuite.ts:285]: Package name policy now handled automatically

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts [hunk 2]
```diff
@@ -166,15 +166,37 @@ export function getFilteredParent(apiItem: ApiItem): ApiItem | undefined {
 }
 
 /**
- * Adjusts the name of the item as needed.
- * Accounts for method overloads by adding a suffix such as "myMethod_2".
+ * Converts invalid filename characters to underscores.
+ */
+function getFileSafeNameForName(apiItemName: string): string {
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:171]: Code that was previously using this helper directly were (incorrectly) not getting the qualified naming for cases like overloads.

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts [hunk 3]
```diff
@@ -166,15 +166,37 @@ export function getFilteredParent(apiItem: ApiItem): ApiItem | undefined {
...
  * @param apiItem - The API item for which the qualified name is being queried.
  *
  * @public
  */
-export function getQualifiedApiItemName(apiItem: ApiItem): string {
-	let qualifiedName: string = getSafeFilenameForName(apiItem.displayName);
+export function getFileSafeNameForApiItem(apiItem: ApiItem): string {
+	const unqualifiedName: string =
+		apiItem.kind === ApiItemKind.Package
+			? getFileSafeNameForName(getUnscopedPackageName(apiItem as ApiPackage))
```
Comment (alexvy86) [inline tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:196]: Curious if there was a big reason why we unscope the package name. Feels like at least in some cases the scope could still be useful in the filename-safe output of the function.

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts [hunk 4]
```diff
@@ -166,15 +166,37 @@ export function getFilteredParent(apiItem: ApiItem): ApiItem | undefined {
...
+ * - Accounts for method overloads by adding a suffix (such as "myMethod_2").
  *
  * @param apiItem - The API item for which the qualified name is being queried.
  *
  * @public
  */
-export function getQualifiedApiItemName(apiItem: ApiItem): string {
-	let qualifiedName: string = getSafeFilenameForName(apiItem.displayName);
+export function getFileSafeNameForApiItem(apiItem: ApiItem): string {
+	const unqualifiedName: string =
```
Comment (alexvy86) [inline tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:194]: Just do away with `unqualifiedName`? Its only use is being assigned to `qualifiedName`.

**Review by alexvy86 — APPROVED** [review]
One small thing, lgtm otherwise

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts (hunk 4 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:194]: I actually originally had it that way and changed it, haha. Might just be me being overly pedantic, but I didn't want to assign the "unqualified" name to something called "qualified".

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts (hunk 3 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:196]: Huh. You know, that's fair. We would still need to omit the `/` between the scope and the package name in those cases, but the `getFileSafeNameForName` helper would handle that for us.

I still think the _default_ behavior probably wants to be to omit the scope, but there's no reason to enforce that. I'll update this.

Comment (Josmithr) [reply tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts]: Updated

**Review by alexvy86 — APPROVED** [review]
🚀

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts (hunk 4 above)
Comment (alexvy86) [reply tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts]: I think I'd have used `result` in that case 😄 .

Comment (Josmithr) [reply tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts]: Fair

**Pull request merged**